_posts/2017-02-23-the-component-life.md

```diff
@@ -0,0 +1,118 @@
+---
+layout: post
+title: "The Component Life"
+date: 2017-02-23
+tags: 
+- Web components
+- Javascript
+categories: 
+- javascript
+- frontend
+twitter_text: "The Component Life #webcomponents #javascript"
+authors: Nathan White
+image: /images/modules.jpg
+image_credit: Nathan White
+external_links_target_blank: true
+---
+
+At RimDev we're trying to rethink the way we build our frontend. We have so many properties with common functionality, and it was starting to be cumbersome (not to mention a waste of time) to implement similar features across all of our products. We knew there was a better approach. This is what brought us to a redistributable component architecure. 
+
+One of the recent components we've built is for our "Related Content." Now on the backend, it's just one simple endpoint. However, on the frontend, we have many different contexts for where/why this component is used. For example, sometimes we might utilize this component on a static blog site. Other times we might use it to show related articles inside of one of our products. We needed to build out these pieces in a very extensible manner for the frontend. There should be a minimal barrier to implementation and it should allow the frontend developer(designer) an easy workflow to embed inside another site.
```

Comment:
Suggestion revision: "allow the frontend developer(designer)" -> "allow a frontend developer or designer".